tower-http/src/content_encoding.rs

```diff
@@ -565,4 +627,31 @@ mod tests {
         let encoding = Encoding::from_headers(&headers, SupportedEncodingsAll::default());
         assert_eq!(Encoding::Identity, encoding);
     }
+
+    #[test]
+    fn accept_encoding_header_with_wildcard() {
+        let mut headers = http::HeaderMap::new();
+        headers.append(
+            http::header::ACCEPT_ENCODING,
+            http::HeaderValue::from_static("*;q=0.8,gzip;q=0.5"),
+        );
+        let encoding = Encoding::from_headers(&headers, SupportedEncodingsAll::default());
+        assert_eq!(Encoding::Identity, encoding);
```

Comment:
I'm not sure we want to use identity here as the first option since we should be able to provide another compression algorithm other than gzip and still follow the specification (if the server supports it). 

> The asterisk "*" symbol in an Accept-Encoding field matches any available content-coding not explicitly listed in the header field.

Which should mean brotli for example is acceptable in this case. Forgive me if this was what you did before my previous comments on this pr because in that case I agree we should change the order we select compression algorithm to prevent situations like this.